causalpy/skl_experiments.py

```diff
@@ -362,14 +375,26 @@ def __init__(
         treatment_threshold,
         model=None,
         running_variable_name="x",
+        epsilon: float = 0.001,
+        bandwidth: Optional[float] = None,
         **kwargs,
     ):
         super().__init__(model=model, **kwargs)
         self.data = data
         self.formula = formula
         self.running_variable_name = running_variable_name
         self.treatment_threshold = treatment_threshold
-        y, X = dmatrices(formula, self.data)
+        self.bandwidth = bandwidth
+        self.epsilon = epsilon
+
+        if self.bandwidth is not None:
+            fmin = self.treatment_threshold - self.bandwidth
+            fmax = self.treatment_threshold + self.bandwidth
+            filtered_data = self.data.query(f"{fmin} <= x <= {fmax}")
```

Comment:
Should we raise a warning when the bandwidth parameter makes the training set very small? Say less than 10 points?